src/explore-education-statistics-admin/src/pages/release/edit-release/data/ReleaseDataUploadsSection.tsx

```diff
@@ -146,15 +148,38 @@ const ReleaseDataUploadsSection = ({ publicationId, releaseId }: Props) => {
   };
 
   const handleSubmit = useFormSubmit<FormValues>(async (values, actions) => {
+    setIsUploading(true);
     await editReleaseDataService.uploadDataFiles(releaseId, {
       subjectTitle: values.subjectTitle,
       dataFile: values.dataFile as File,
       metadataFile: values.metadataFile as File,
     });
-
+    setIsUploading(false);
     await resetPage(actions);
   }, errorCodeMappings);
 
+  const handleDelete = async (
+    dataFileToDelete: DeleteDataFile,
+    form: FormikActions<{}>,
+  ) => {
+    const files = [...dataFiles];
+    const fileToDelete = files.find(
+      file => file.filename === dataFileToDelete.file.filename,
+    );
+
+    if (!fileToDelete) {
+      return;
+    }
+    fileToDelete.isDeleting = true;
```

Comment:
This is a potential bug because we're directly mutating state rather than setting the new state in an immutable way. Direct mutations are bad because they don't let React know that something has changed so it doesn't know to perform a re-render.

On another note, it's weird that we're mutating a reference to one of `dataFiles` via the `files` variable (which is redundant).

Probably the only reason you're getting away with this currently is because `setDeleteDataFile` is called straight afterwards (which tells React to re-render), but this would probably break if you ever took that out.

I would recommend changing it to call `setFiles` with the new state constructed immutably, perhaps using `.map`.

Reply:
yeah see what you mean Nick - cheers for that, will change.

Reply:
done